twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -380,45 +380,53 @@ public function authenticate($username, $password) {
         $errorCode = 0;
         $res = 0;
 
-        try {
-            $client = $this->httpClientService->newClient();
-            $res = $client->get($url, $options);
-            $body = $res->getBody();
-            $body = json_decode($body);
+        if ($url) {
```

Comment:
Thanks! Hm, I can't think of a situation in which we should send an empty password, so I think this is fine -- @cornelinux can you think of one?

Reply:
@fredreichbier ???
An empty PIN or an empty OTP value?

Reply:
Yes -- I guess the question is: If a user leaves the "enter OTP" field blank and clicks submit, should we send a ``/validate/check`` request with an empty password to privacyIDEA, or should we not send a request to privacyIDEA at all?

Reply:
Basically I think why should it be possible to send an empty password. I am not decided. I think it might make sense to require a password, since at this point it probably makes no sense?
(Unlesse the user has a spass token)

Reply:
We are already requiring a password in the client-side validation. Hence the comment, that this is a situation that should never occur, if the client is well behaved. If there is any reason to do a single-shot authentication without a password, it will require changing the code for the client. This PR has no consequence for a user using the owncloud webclient to sign in, since it only changes behavior for requests that the webclient will not make.

Reply:
OK, sounds good! Thank you for your input and your patience :)